vault/resource_kmip_secret_role.go

```diff
@@ -0,0 +1,312 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+const (
+	fieldOperationActivate         = "operation_activate"
+	fieldOperationAddAttribute     = "operation_add_attribute"
+	fieldOperationAll              = "operation_all"
+	fieldOperationCreate           = "operation_create"
+	fieldOperationDestroy          = "operation_destroy"
+	fieldOperationDiscoverVersions = "operation_discover_versions"
+	fieldOperationGet              = "operation_get"
+	fieldOperationGetAttributeList = "operation_get_attribute_list"
+	fieldOperationGetAttributes    = "operation_get_attributes"
+	fieldOperationLocate           = "operation_locate"
+	fieldOperationNone             = "operation_none"
+	fieldOperationRegister         = "operation_register"
+	fieldOperationRekey            = "operation_rekey"
+	fieldOperationRevoke           = "operation_revoke"
+)
+
+var kmipRoleAPIBooleanFields = []string{
+	fieldOperationActivate,
+	fieldOperationAddAttribute,
+	fieldOperationAll,
+	fieldOperationCreate,
+	fieldOperationDestroy,
+	fieldOperationDiscoverVersions,
+	fieldOperationGet,
+	fieldOperationGetAttributeList,
+	fieldOperationGetAttributes,
+	fieldOperationLocate,
+	fieldOperationNone,
+	fieldOperationRegister,
+	fieldOperationRekey,
+	fieldOperationRevoke,
+}
+
+func kmipSecretRoleResource() *schema.Resource {
+	return &schema.Resource{
+		Create: kmipSecretRoleCreate,
+		Read:   kmipSecretRoleRead,
+		Update: kmipSecretRoleUpdate,
+		Delete: kmipSecretRoleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"path": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Path where KMIP backend is mounted",
+				ValidateFunc: validateNoTrailingLeadingSlashes,
+			},
+			"scope": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the scope",
+			},
+			"role": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the role",
+			},
+			"tls_client_key_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Client certificate key type, rsa or ec",
+			},
+			"tls_client_key_bits": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "Client certificate key bits, valid values depend on key type",
+			},
+			"tls_client_ttl": {
```

Comment:
For consistency, we can probably factor these keys out to constants like you are doing below. Then update any other functions to use them.

Reply:
Added constants in ab39eb6